### PR TITLE
First Terra Commit

### PR DESCRIPTION
**Pull request by Ramesh1984-10**

### CODE REVIEW

File: VPC/module/Sg.tf [hunk 1]
```diff
@@ -0,0 +1,50 @@
+# // Create Security Group for Baiston Host 
+
+# resource "aws_security_group" "ssh-security-group" {
```
Comment (b44rawat) [inline VPC/module/Sg.tf:3]: why commented code present here?

File: VPC/module/main.tf [hunk 2]
```diff
@@ -0,0 +1,189 @@
+resource "aws_vpc" "main" {
+    cidr_block = var.vpc_cidr
+    instance_tenancy = "default"
+    enable_dns_hostnames = true
+  tags = {
+    "Name" = "MY-VPC"
```
Comment (b44rawat) [inline VPC/module/main.tf:6]: why hardcoding name inside the module?

File: VPC/module/main.tf [hunk 3]
```diff
@@ -0,0 +1,189 @@
+resource "aws_vpc" "main" {
+    cidr_block = var.vpc_cidr
+    instance_tenancy = "default"
```
Comment (b44rawat) [inline VPC/module/main.tf:3]: Why hardcoding parameters?

File: VPC/module/main.tf [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+}
+
+
+
+
+// 2 Private  Subnets Creation
+
+# Private subnet creation
+resource "aws_subnet" "private" {
+  count  = length(var.subnet_tag)
```
Comment (b44rawat) [inline VPC/module/main.tf:67]: Why are you using count mechanism here? instead use for_each

File: main.tf [hunk 5]
```diff
@@ -0,0 +1,13 @@
+# VPC module 
+module "vpc" {
+    source = "./VPC/module"
+    vpc_cidr = var.vpc_cidr
+    public_sub_cidr = var.public_sub_cidr
+    az  = var.az
```
Comment (b44rawat) [inline main.tf:6]: use proper naming convention

File: main.tf [hunk 6]
```diff
@@ -0,0 +1,13 @@
+# VPC module 
+module "vpc" {
+    source = "./VPC/module"
```
Comment (b44rawat) [inline main.tf:3]: location is not proper, use common folder where you can store all the modules.

File: terraform.tfvars [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+db-private_sub_cidr = "192.168.4.0/24"
+
+
+subnet_tag = {
+  #database      = "Database Subnet"
+  application      = "Application Subnet"
+  frontend = "Frontend Subnet"
+}
+
+public-rt-cidr = "0.0.0.0/0"
```
Comment (b44rawat) [inline terraform.tfvars:16]: What's the use-case of this?

**Comment by b44rawat** [comment]
Terraform directory structure is not proper

File: VPC/module/Sg.tf (hunk 1 above)
Comment (Ramesh1984-10) [reply VPC/module/Sg.tf:3]: Resolved

**Pull request closed without merging**